index.js

```diff
@@ -85,23 +87,29 @@ cli.on('respawn', function(flags, child) {
   log.info('Respawned to PID:', pid);
 });
 
+
 function run() {
-  cli.launch({
+  cli.prepare({
     cwd: opts.cwd,
     configPath: opts.gulpfile,
     require: opts.require,
     completion: opts.completion,
-  }, handleArguments);
+  }, function(env) {
+
+    var cfgLoadOrder = ['home', 'cwd'];
+    var cfg = loadConfigFiles(env.configFiles['.gulp'], cfgLoadOrder);
+    opts = mergeConfigToCliFlags(opts, cfg);
+    env = mergeConfigToEnvFlags(env, cfg);
+    env.configProps = cfg;
+
+    this.execute(env, handleArguments);
```

Comment:
should this be `cli` instead of `this`? - I'm thinking that will help when we switch to promises.